commons-math/src/main/java/org/apache/commons/math4/linear/ArrayFieldVector.java

```diff
@@ -0,0 +1,1125 @@
+        }
+        return new ArrayFieldVector<>(field, out, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapAdd(T d) throws NullArgumentException {
+        T[] out = MathArrays.buildArray(field, data.length);
+        for (int i = 0; i < data.length; i++) {
+            out[i] = data[i].add(d);
+        }
+        return new ArrayFieldVector<>(field, out, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapAddToSelf(T d) throws NullArgumentException {
+        for (int i = 0; i < data.length; i++) {
+            data[i] = data[i].add(d);
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapSubtract(T d) throws NullArgumentException {
+        T[] out = MathArrays.buildArray(field, data.length);
+        for (int i = 0; i < data.length; i++) {
+            out[i] = data[i].subtract(d);
+        }
+        return new ArrayFieldVector<>(field, out, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapSubtractToSelf(T d) throws NullArgumentException {
+        for (int i = 0; i < data.length; i++) {
+            data[i] = data[i].subtract(d);
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapMultiply(T d) throws NullArgumentException {
+        T[] out = MathArrays.buildArray(field, data.length);
+        for (int i = 0; i < data.length; i++) {
+            out[i] = data[i].multiply(d);
+        }
+        return new ArrayFieldVector<>(field, out, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapMultiplyToSelf(T d) throws NullArgumentException {
+        for (int i = 0; i < data.length; i++) {
+            data[i] = data[i].multiply(d);
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapDivide(T d)
+        throws NullArgumentException, MathArithmeticException {
+        MathUtils.checkNotNull(d);
+        T[] out = MathArrays.buildArray(field, data.length);
+        for (int i = 0; i < data.length; i++) {
+            out[i] = data[i].divide(d);
+        }
+        return new ArrayFieldVector<>(field, out, false);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapDivideToSelf(T d)
+        throws NullArgumentException, MathArithmeticException {
+        MathUtils.checkNotNull(d);
+        for (int i = 0; i < data.length; i++) {
+            data[i] = data[i].divide(d);
+        }
+        return this;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public FieldVector<T> mapInv() throws MathArithmeticException {
+        T[] out = MathArrays.buildArray(field, data.length);
+        final T one = field.getOne();
+        for (int i = 0; i < data.length; i++) {
+            try {
+                out[i] = one.divide(data[i]);
+            } catch (final MathArithmeticException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)